src/olympia/addons/models.py

```diff
@@ -1260,15 +1260,18 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def has_complete_metadata(self):
-        return all(self.get_required_metadata())
-
-    def get_required_metadata(self):
-        if not self.has_listed_versions():
+    def has_complete_metadata(self, for_channel=None):
+        return all(self.get_required_metadata(for_channel=for_channel))
+
+    def get_required_metadata(self, for_channel=None):
+        if for_channel is None:
+            for_channel = (
+                amo.RELEASE_CHANNEL_LISTED if self.has_listed_versions()
+                else amo.RELEASE_CHANNEL_UNLISTED)
+        if for_channel == amo.RELEASE_CHANNEL_UNLISTED:
             # Add-ons with only unlisted versions have no required metadata.
             return []
-        latest_version = self.find_latest_version(
-            channel=amo.RELEASE_CHANNEL_LISTED)
+        latest_version = self.versions.latest()  # Version.from_upload does it.
```

Comment:
This seems like a fragile hypothesis to build upon. IMHO `Version.from_upload` shouldn't copy over the license information to unlisted versions in the first place... Any reason to not keep the filtering that was done before ?